lib/grape/validations/params_scope.rb

```diff
@@ -140,9 +140,13 @@ def require_optional_fields(context, opts)
       end
 
       def validate_attributes(attrs, opts, &block)
+        # check fail_fast flag
+        fail_fast = opts.delete(:fail_fast) || false
```

Comment:
You should not be modifying `opts` being passed in here, combine the `clone` below.